src/Components/Components/src/Routing/NavLink.cs

```diff
@@ -49,29 +53,38 @@ public class NavLink : IComponent, IDisposable
         [Inject] private IUriHelper UriHelper { get; set; }
 
         /// <inheritdoc />
-        public void Configure(RenderHandle renderHandle)
+        protected override void OnInit()
         {
-            _renderHandle = renderHandle;
-
             // We'll consider re-rendering on each location change
             UriHelper.OnLocationChanged += OnLocationChanged;
         }
 
         /// <inheritdoc />
-        public Task SetParametersAsync(ParameterCollection parameters)
+        public override Task SetParametersAsync(ParameterCollection parameters)
         {
-            // Capture the parameters we want to do special things with, plus all as a dictionary
-            parameters.TryGetValue(RenderTreeBuilder.ChildContent, out _childContent);
-            parameters.TryGetValue("class", out _cssClass);
-            parameters.TryGetValue("href", out string href);
-            ActiveClass = parameters.GetValueOrDefault(nameof(ActiveClass), DefaultActiveClass);
-            Match = parameters.GetValueOrDefault(nameof(Match), NavLinkMatch.Prefix);
-            _allAttributes = parameters.ToDictionary();
-
-            // Update computed state and render
+            // Run the default parameter setting logic to populate our properties.
+            base.SetParametersAsync(parameters);
+
+            // Update computed state
+            var href = (string)null;
+            if (AdditionalAttributes != null && AdditionalAttributes.TryGetValue("href", out var obj))
+            {
+                href = Convert.ToString(obj);
+            }
+
             _hrefAbsolute = href == null ? null : UriHelper.ToAbsoluteUri(href).AbsoluteUri;
             _isActive = ShouldMatch(UriHelper.GetAbsoluteUri());
-            _renderHandle.Render(Render);
+
+            ActiveClass ??= DefaultActiveClass;
```

Comment:
Rather than writing to `ActiveClass`, could we combine them on some other local (e.g., `var effectiveActiveClass = ActiveClass ?? DefaultActiveClass`)?

Because, although we don't really expect people to be reading back the values of parameters on component instances, the API does allow them to do so, and then someone might start depending on this behavior which is more of an implementation detail.

Reply:
I took this suggestion and just folded this expression into the one place where it's used.